File: Documentation/5.4/Samples/python/ClientApi/Session/Configuration/CustomizeCollectionAssignmentForEntities.py
from ravendb import DocumentStore
from ravendb.documents.conventions import DocumentConventions

from examples_base import ExampleBase, Category


class CustomizeCollectionAssignmentForEntities(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_collection_assignment_for_entities(self):
        store = DocumentStore()

        # region custom_collection_name
        def __find_collection_name(object_type: type) -> str:
            if issubclass(object_type, Category):
                return "ProductGroups"

            return DocumentConventions.default_get_collection_name(object_type)

        store.conventions.find_collection_name = __find_collection_name
        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Session/Configuration/CustomizeCollectionAssignmentForEntities.php
<?php

use RavenDB\Documents\Conventions\DocumentConventions;
use RavenDB\Documents\DocumentStore;
use RavenDB\Samples\Infrastructure\Orders\Category;

class CustomizeCollectionAssignmentForEntities
{
    public function example(): void
    {
        $store = new DocumentStore();

        # region custom_collection_name
        $store->getConventions()->setFindCollectionName(function($className) {
            if (is_subclass_of($className, Category::class)) {
                return "ProductGroups";
            }

            return DocumentConventions::defaultGetCollectionName($className);
        });
        # endregion
    }
}
